src/cbc_sdk/platform/admin_grants.py

```diff
@@ -0,0 +1,446 @@
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            **kwargs (dict): Not used, retained for compatibility.
+
+        Returns:
+            GrantQuery: The query object for this alert type.
+        """
+        return GrantQuery(cls, cb)
+
+    def _refresh(self):
+        """
+        Rereads the grant data from the server.
+
+        Returns:
+            bool: True if refresh was successful, False if not.
+        """
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id)
+        resp = self._cb.get_object(url)
+        self._info = resp
+        self._profiles = [Grant.Profile(self._cb, self, prof.get('profile_uuid', None), prof)
+                          for prof in resp.get('profiles', [])]
+        self._last_refresh_time = time.time()
+        return True
+
+    def _update_object(self):
+        """
+        Updates the grant data on the server.
+
+        Returns:
+            str: The principal URN of this grant object.
+        """
+        save_info = copy.deepcopy(self._info)
+        if 'principal' not in save_info:
+            raise ApiError("principal must be specified for a Grant object")
+        save_info['profiles'] = [copy.deepcopy(prof._info) for prof in self._profiles if 'profile_uuid' in prof._info]
+
+        if all([key in save_info for key in ['created_by', 'updated_by', 'create_time', 'update_time']]):
+            log.debug("Updating the Grant object for principal {0:s}".format(save_info['principal']))
+            url = self.urlobject_single.format(self._cb.credentials.org_key, save_info['principal'])
+            ret = self._cb.put_object(url, save_info)
+        else:
+            log.debug("Creating a new Grant object for principal {0:s}".format(save_info['principal']))
+            url = self.urlobject.format(self._cb.credentials.org_key)
+            ret = self._cb.post_object(url, save_info)
+        update_return = self._refresh_if_needed(ret)
+
+        # add all profiles not yet part of grant
+        for profile in self._profiles:
+            if 'profile_uuid' not in profile._info:
+                profile._update_object()
+        return update_return
+
+    def _delete_object(self):
+        """Deletes the grant from the principal."""
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id)
+        ret = self._cb.delete_object(url)
+        self._refresh_if_needed(ret)
+
+    @classmethod
+    def create(cls, cb, principal):
+        """
+        Create a new grant object for the specified principal.
+
+        Args:
+            cb (CBCloudAPI): A reference to the CBCloudAPI object.
+            principal (str): The principal URN for the new grant object.
+
+        Returns:
+            Grant: The new grant object.
+        """
+        grant = Grant(cb, principal, {'principal': principal})
+        grant._full_init = True
```

Comment:
This is not correct and misleads the current object. The Grant is not `_full_init`

Reply:
If I take it out, two of my tests fail with the following error:
```
Failed: GET called for /access/v2/orgs/test/grants/psc:user:ABC12345:DEF67890 when it shouldn't be
```
There's a sneak path in the `__setattr__` code in the `MutableBaseModel` that is causing `refresh()` to be called when it shouldn't be.

Reply:
Then we should implement an alternative approach to solve this issue. Let's not mess with existing practices as this may cause more issues in the future

Reply:
What do you suggest? I was racking my brains for _hours_ to find an alternate way of solving this, to no avail.

Reply:
I see three options with varying types of UX:
* Require the entire grant like we have done for reputation https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/develop/src/cbc_sdk/platform/reputation.py#L93
* Create a Builder like we have for Profiles
* Lastly, if the following code is the experience we desire then we need to add a new property similar to _full_init that will allow for additional new behavior in `def _set(self, attrname, new_value)`. A possible name could be `_created`
```
grant = Grant.create(api, 'psc:user:ABC12345:DEF67890')
grant.roles = ["psc:role::SECOPS_ROLE_MANAGER"]
grant.org_ref = 'psc:org:test'
grant.principal_name = 'Doug Jones'
```